pkg/referrerstore/oras/oraswithcache.go

```diff
@@ -0,0 +1,133 @@
+package oras
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/deislabs/ratify/pkg/common"
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	"github.com/deislabs/ratify/pkg/referrerstore/config"
+	"github.com/dgraph-io/ristretto"
+	"github.com/dgraph-io/ristretto/z"
+	"github.com/opencontainers/go-digest"
+	"github.com/sirupsen/logrus"
+)
+
+type Operation int
+
+const (
+	Name Operation = iota
+	GetConfig
+	ListReferrers
+	GetBlobContent
+	GetReferenceManifest
+	GetSubjectDescriptor
+)
+
+type orasStoreWithInMemoryCache struct {
+	base      referrerstore.ReferrerStore
+	cache     *ristretto.Cache
```

Comment:
ristretto implements its own GC for expired entries. The GC rate is `2.5s`. Will it stop the world if there are lots of cache entries?

Reply:
good question! ristretto uses sharding to help reduce gc pause, but it's possible that too many entries might incur large GC pause time. There is another library: [BigCache](https://github.com/allegro/bigcache) which optimized the bottom map to avoid expensive GC cycles. But its average performance is worse than ristretto according to https://github.com/dgraph-io/ristretto#Benchmarks